projects/online_sequence_classification/clustering/clustering_interface.py

```diff
@@ -0,0 +1,275 @@
+import heapq
```

Comment:
You might look at Python's builtin PriorityQueue module. It is built on top of `heapq` and is usually an easier interface to use.